build/gen_dev_docker.sh

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bash
+
+set -e
+
+DIR_UUID=$(cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 32 | head -n 1)
+user=$(id -un)
+group=$(id -gn)
+uid=$(id -u)
+gid=$(id -g)
+tmpdir="/tmp/fdb-docker-${DIR_UUID}"
+image=fdb-dev
+
+pushd .
+mkdir ${tmpdir}
+cd ${tmpdir}
+
+cat <<EOF >> Dockerfile
+FROM foundationdb/foundationdb-build:latest
+RUN groupadd -g ${gid} ${group} && useradd -u ${uid} -g ${gid} -m ${user}
+
+USER ${user}
+CMD scl enable devtoolset-8 rh-python36 rh-ruby24 -- bash
+
+EOF
+
+echo "Created ${tmpdir}"
+echo "Buidling Docker container ${image}"
+sudo docker build -t ${image} .
+
+popd
+
+echo "Writing startup script"
+mkdir -p $HOME/bin
+cat <<EOF > $HOME/bin/fdb-dev
+#!/usr/bin/bash
+
+sudo docker run --rm `# delete (temporary) image after return` \\
+                -it `# Run in interactive mode and simulate a TTY` \\
+                --privileged=true `# Run in privileged mode ` \\
+                --cap-add=SYS_PTRACE \\
+                --security-opt seccomp=unconfined \\
+                -v '/home:/home' `# Mount home directory` \\
```

Comment:
```suggestion
                -v "$HOME:/home" `# Mount home directory` \\
```

Reply:
@jzhou77 Is the motivation for you suggestion so that it works on OSX too? I think that makes sense

Reply:
Right. On Mac, the standard home is at `/Users/username`.

Reply:
Running this on MacOS is a horrible idea (the file system is so slow that compiling FDB will probably take three days or so...).

But I guess this change could still make sense for non-standard systems. Shall I do what @jzhou77 suggests or would `-v${HOME}:${HOME}` be better? Otherwise stuff like `compile_commands.json` will not work from outside the docker container.

Reply:
> Otherwise stuff like compile_commands.json will not work from outside the docker container.

Good point, I agree `-v${HOME}:${HOME}` would be better